tests/ttnn/unit_tests/gtests/test_repeat_interleave.cpp

```diff
@@ -0,0 +1,105 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "gtest/gtest.h"
+
+#include "tt_metal/common/bfloat16.hpp"
+#include "ttnn/device.hpp"
+#include "ttnn/operations/core.hpp"
+#include "ttnn/async_runtime.hpp"
+#include "ttnn/operations/data_movement.hpp"
+#include "tt_numpy/functions.hpp"
+#include "tt_metal/common/logger.hpp"
+
+#include "ttnn_test_fixtures.hpp"
+
+#include <memory>
+
+namespace ttnn {
+namespace operations {
+namespace data_movement {
+namespace test {
+
+void run_repeat_interleave_test(tt::tt_metal::Device* device, const uint32_t repeats, const uint32_t dim) {
```

Comment:
This is not the best way to write op test. @tt-asaigal directed to `tests/tt_eager/ops/test_bmm_op.cpp`, but I wanted to explore this path to better understand whats going on under the hood

Reply:
I think @eyonland had an template example of a `ttnn` unit test